src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2761,15 +2774,19 @@ Handle java_lang_Throwable::create_initialization_error(JavaThread* current, Han
   assert(throwable.not_null(), "shouldn't be");
 
   // Now create the message from the original exception and thread name.
-  Symbol* message = java_lang_Throwable::detail_message(throwable());
   ResourceMark rm(current);
+  const char *message = nullptr;
+  oop detailed_message = java_lang_Throwable::message(throwable());
+  if (detailed_message != nullptr) {
+    message = java_lang_String::as_utf8_string(detailed_message);
+  }
```

Comment:
It looks like you did remove converting the message to a Symbol here, so are there other places where long symbol names come through this code?  We should catch them at their sources.

Reply:
> It looks like you did remove converting the message to a Symbol here, so are there other places where long symbol names come through this code? We should catch them at their sources.

Please see https://github.com/openjdk/jdk/pull/14938#issuecomment-1644534913. Exceptions::log_exception can be simply changed to use `char *`. Other callers might need a bit more work.